model/src/main/kotlin/OrtResult.kt

```diff
@@ -400,6 +403,23 @@ data class OrtResult(
     fun isResolved(vulnerability: Vulnerability): Boolean =
         getResolutions().vulnerabilities.any { it.matches(vulnerability) }
 
+    /**
+     * Return the resolutions matching [issue].
+     */
+    fun getResolutionsFor(issue: Issue): List<IssueResolution> = getResolutions().issues.filter { it.matches(issue) }
+
+    /**
+     * Return the resolutions matching [ruleViolation].
+     */
+    fun getResolutionsFor(ruleViolation: RuleViolation): List<RuleViolationResolution> =
+        getResolutions().ruleViolations.filter { it.matches(ruleViolation) }
+
+    /**
+     * Return the resolutions matching [vulnerability].
+     */
+    fun getResolutionsFor(vulnerability: Vulnerability): List<VulnerabilityResolution> =
+        getResolutions().vulnerabilities.filter { it.matches(vulnerability) }
+
```

Comment:
Yes, I believe that's roughly capturing it. Certainly I also notice that there will be further clean-ups in the area of `DefaultResolutionProvider` possible on top. I'm not sure what you thinking behing `ResolutionProvider` for `OrtResult`s is, but I also guess it's not needed discussing this now.

Reply:
> I'm not sure what you thinking behing `ResolutionProvider` for `OrtResult`s is, but I also guess it's not needed discussing this now.

Well, I believe it's worth discussion this at least briefly now, because your branch name of "reporter-remove-resolution-provider" sounds as if you'd want to really remove the `ResolutionProvider` interface, whereas my idea would be to have `data class OrtResult : ResolutionProvider {}`.

Reply:
> Well, I believe it's worth discussion this at least briefly now, because your branch name

Sure, then we have a reason to do so. (The branch name does not matter, I don't plan to remove that interface)

Reply:
Ok. But since you're already adding all functions required to implement `ResolutionProvider` to `OrtResult`, just partly with different names, wouldn't it make sense to use the exact same names / signatures as in `ResolutionProvider` and indeed make `OrtResult` implement the `ResolutionProvider` interface?

Reply:
I would like to understand the need for this, e.g. why is it beneficial?

Reply:
It's IMO just cleaner code. You're basically adding all the functions mentioned in `ResolutionProvider` to `OrtResult`, effectively making the `OrtResult` a `ResolutionProvider`, but you're not calling it a `ResolutionProvider`. Why shouldn't you, if you say you're not intending to remove `ResolutionProvider`?

Reply:
And the use-case would be that `OrtResult` could be used everywhere where a `ResolutionProvider` is required.

Reply:
> It's IMO just cleaner code. 

I believe there should be a reason / use case to implement the interface. If there is no, then I prefer to not do it in scope of this PR. Most importantly because it comes at a cost.

> And the use-case would be that OrtResult could be used everywhere where a ResolutionProvider is required.

And what would the use case be which needs this? (I don't see any)

Reply:
> Most importantly because it comes at a cost.

Which cost? You just need to add ": ResolutionProvider" to the class declaration of `OrtResult` and rename a few functions. What am I missing?

> I still preferred if this topic was left out of scope of this PR, if you're fine with it.

To me, this is the main deciding factor of this PR. I only see two options currently:

- Either `OrtResult` has all properties of a `ResolutionProvider` and should thus implement that interface, or
- We remove the `ResolutionProvider` interface as we (maybe?) don't need it going forward as the `OrtResult` becomes the only provider (besides the trivial `File` provider which could be modeled simply as as `Resolutions`).

Reply:
I feel that this discussion is getting more complex and fundamental. Shall we try to resolve it in the next Kotlin developer meeting?

Reply:
Yes, makes sense. I've put it onto the agenda already.

Reply:
I've made it implement the interface.